### PR TITLE
CASMINST-6830 1.5.1 Install Upgrade Docs

### PR DESCRIPTION
**Pull request by juliansmit-hpe**
# Description

Upgrade documentation from CSM 1.5.0 to 1.5.1.
Relates to: https://jira-pro.it.hpe.com:8443/browse/CASMINST-6830
# Checklist

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

File: upgrade/1.5.1/index.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+This document guides an administrator through the patch update to Cray Systems Management `v1.5.1` from `1.5.0`. 
+If upgrading from CSM `v1.4.x` directly to `v1.5.1`, follow the procedures described
+in [Upgrade CSM](../README.md) instead.
+
+## Bug fixes and improvements
+
+   * Updated IMS to allow for remote node builds
+   * Support for Paradise hardware
+   * Fixed an issue where hms-discovery would put default creds into vault disabling non default credentials
+   * Fixed an issue where crap-ipxe generated scripts would cause boot errors
```
Comment (rfrost-hpe) [inline upgrade/1.5.1/index.md]: This looks wrong. Crap should be cray?

File: upgrade/1.5.1/index.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+* [Steps](#steps)
+
+## Introduction
+
+This document guides an administrator through the patch update to Cray Systems Management `v1.5.1` from `1.5.0`. 
+If upgrading from CSM `v1.4.x` directly to `v1.5.1`, follow the procedures described
+in [Upgrade CSM](../README.md) instead.
+
+## Bug fixes and improvements
+
```
Comment (rfrost-hpe) [inline upgrade/1.5.1/index.md]: @ndavidson-hpe how do we communicate your hot fix that went into 1.5.0 for CASMINST-6715 is now in this release?  What problem does it actually fix?

File: upgrade/1.5.1/index.md
Comment (rustydb) [inline upgrade/1.5.1/index.md:1]: Rename file to `README.md`, make `index.md` a symlink to `README.md`.

**Review by rustydb — APPROVED** [review]

File: upgrade/1.5.1/index.md (hunk 2 above)
Comment (ndavidson-hpe) [reply upgrade/1.5.1/index.md]: It fixes an issue where an old version of the spire-agent RPM was installed to Nexus.

Comment (rfrost-hpe) [reply upgrade/1.5.1/index.md]: but what text should be  used to communicate this to the customer? Link the ticket? Link the hot fix information?

Comment (ndavidson-hpe) [reply upgrade/1.5.1/index.md]: Probably a link to the hotfix information. Not sure how much customer can see internal triage tickets.

File: upgrade/1.5.1/README.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+## Introduction
+
+This document guides an administrator through the patch update to Cray Systems Management `v1.5.1` from `1.5.0`.
+If upgrading from CSM `v1.4.x` directly to `v1.5.1`, follow the procedures described
+in [Upgrade CSM](../README.md) instead.
+
+## Bug fixes and improvements
+
+* Updated IMS to allow for remote node builds
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Updated [IMS](../../glossary.md#image-management-service-ims) to allow for remote node builds
```

File: upgrade/1.5.1/README.md [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+* Support for Paradise hardware
+* Fixed an issue where `hms-discovery` would put default credentials into vault disabling non default credentials
+* Fixed an issue where `cray-ipxe` generated scripts would cause boot errors
+* Fixed an issue where `BOS v2` would send large queries to CFS resulting 503 and 431 responses
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Fixed an issue where [BOS](../../glossary.md#boot-orchestration-service-bos) v2 would send large queries to [CFS](../../glossary.md#configuration-framework-service-cfs) resulting 503 and 431 responses
```

File: upgrade/1.5.1/README.md [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+* Fixed an issue where PCS was adding invalid components to power operations
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Fixed an issue where [PCS](../../glossary.md#power-control-service-pcs) was adding invalid components to power operations
```

File: upgrade/1.5.1/README.md [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+* Fixed an issue where IUF would encounter a race condition and stall on transitioning to new stages
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Fixed an issue where [IUF](../../glossary.md#install-and-upgrade-framework-iuf) would encounter a race condition and stall on transitioning to new stages
```

File: upgrade/1.5.1/README.md [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+* Fixed an issue where `Thanos` service is configured without storage limits
+* Fixed an issue where `cray-dns-unbound-manager` `stderr` handling can corrupt configuration
+* Fixed an issue where "sat status" unnecessarily queries BOS for session template for every component
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Fixed an issue where [SAT](../../glossary.md#system-admin-toolkit-sat) `status` unnecessarily queries BOS for session template for every component
```

File: upgrade/1.5.1/README.md [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+* Fixed an issue where a PATCH to a BOS v2 session to change it's name results in a bad state
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Fixed an issue where a PATCH to a BOS v2 session to change its name results in a bad state
```

File: upgrade/1.5.1/README.md [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+* Updated `cray-hms-rts-init` job to include a TTL
+* Updated node-exporter config to monitor `snmp` counters
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Updated `node-exporter` configuration to monitor SNMP counters
```

File: upgrade/1.5.1/README.md [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+* Updated docs to cover switch configuration for NCNs
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Updated documentation to cover switch configuration for [NCNs](../../glossary.md#non-compute-node-ncn)
```

File: upgrade/1.5.1/README.md [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+* Fixed an issue where a PATCH call to BOS v2 components with a filter and non existent component ID would result in 503 and 431 errors
+* Fixed an issue where BOS operators would output errors when all nodes exceed retry limit
+* Fixed an issue where `cray-upload-recovery-images` fails to upload recovery firmware
+* Fixed an issue where "cm health report slingshot refresh" `cmd` was giving a trace back error
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Fixed an issue where `cm health report slingshot refresh` command was giving a trace back error
```

File: upgrade/1.5.1/README.md [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+* Updated docs for IUF to include rolling reboots after upgrading HSN NIC firmware
+* Updated docs to provide instructions for creating a new Nexus repo and adding `rpms`
+* Updated docs for CAPMC to warn about URL character limits
+* Fixed an issue where BOS v2 requests for too many nodes from PCS exceeding the URL character limit
+* Updated docs script `nexus_export_and_restore` to add better checks for determining existence of `nexus-bak` PVC or nexus-backup job
+* Updated docs for `fix_failed_to_start_etcd_on_master` to better specify how to add a process
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
* Updated documentation for IUF to include rolling reboots after upgrading [HSN](../../glossary.md#high-speed-network-hsn) NIC firmware
* Updated documentation to provide instructions for creating a new Nexus repository and adding RPMs
* Updated documentation for [CAPMC](../../glossary.md#cray-advanced-platform-monitoring-and-control-capmc) to warn about URL character limits
* Fixed an issue where BOS v2 requests for too many nodes from PCS exceeding the URL character limit
* Updated Nexus export and restore script to add better checks for determining existence of `nexus-bak` PVC or `nexus-backup` job
* Updated documentation for `fix_failed_to_start_etcd_on_master` to better specify how to add a process
```

File: upgrade/1.5.1/README.md [hunk 13]
```diff
@@ -0,0 +1,316 @@
...
+* Fixed an issue where `backup_smd_postgres.sh` script is not executable
+* `CAST-34869`: Fixed an issue where `cray-hms-rts-snmp` would enter encounter a segmentation fault
+* `CAST-34141`: Fixed an issue where the `QLogic` driver would crash the system
+
+## Steps
+
+1. [CSM 1.5.1 Patch Installation Instructions](#csm-151-patch-installation-instructions)
+1. [Introduction](#introduction)
+1. [Bug fixes and improvements](#bug-fixes-and-improvements)
+1. [Steps](#steps)
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
```

File: upgrade/1.5.1/README.md [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+   INFO: Querying CFS configurations for the following NCNs: x3000c0s5b0n0, ...
+   INFO: Found configuration "management-csm-1.4.0" for component x3000c0s5b0n0
+   ...
+   INFO: Updating existing layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml
+   INFO: Property "commit" of layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml updated ...
+   INFO: Property "name" of layer with repo path /vcs/cray/csm-config-management.git and playbook site.yml updated ...
+   INFO: No layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml found.
+   INFO: Adding a layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml to the end.
+   INFO: Successfully saved CFS configuration "management-csm-1.4.0-backup-20230918T205149"
+   INFO: Successfully saved CFS configuration "management-csm-1.4.0"
```
Comment (mharding-hpe) [inline upgrade/1.5.1/README.md]: Ideally we should just get actual output from an upgrade, but failing that, this makes the changes that I can anticipate:

```suggestion
   INFO: Found configuration "management-csm-1.5.0" for component x3000c0s5b0n0
   ...
   INFO: Updating existing layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml
   INFO: Property "commit" of layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml updated ...
   INFO: Property "name" of layer with repo path /vcs/cray/csm-config-management.git and playbook ncn_nodes.yml updated ...
   INFO: No layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml found.
   INFO: Adding a layer with repo path /vcs/cray/csm-config-management.git and playbook ncn-initrd.yml to the end.
   INFO: Successfully saved CFS configuration "management-csm-1.5.0-backup-20240410T205149"
   INFO: Successfully saved CFS configuration "management-csm-1.5.0"
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Mostly suggesting some pointers into the glossary, along with some minor linting.

File: upgrade/1.5.1/index.md (hunk 2 above)
Comment (rfrost-hpe) [reply upgrade/1.5.1/index.md]: @ndavidson-hpe did you come up with text on what can be added here?

File: upgrade/1.5.1/README.md [hunk 15]
```diff
@@ -0,0 +1,312 @@
...
+* Updated documentation to provide instructions for creating a new Nexus repository and adding RPMs
+* Updated documentation for [CAPMC](../../glossary.md#cray-advanced-platform-monitoring-and-control-capmc) to warn about URL character limits
+* Fixed an issue where BOS v2 requests for too many nodes from PCS exceeding the URL character limit
+* Updated Nexus export and restore script to add better checks for determining existence of `nexus-bak` PVC or `nexus-backup` job
+* Updated documentation for `fix_failed_to_start_etcd_on_master` to better specify how to add a process
+* Fixed an issue with IUF where process-media/pre-install-check dislikes PDF file
+* Fixed an issue where `backup_smd_postgres.sh` script is not executable
+* `CAST-34869`: Fixed an issue where `cray-hms-rts-snmp` would enter encounter a segmentation fault
+* `CAST-34141`: Fixed an issue where the `QLogic` driver would crash the system
+
```
Comment (ndavidson-hpe) [inline upgrade/1.5.1/README.md:43]: Add a note about fix for spire-agent version issue fixed with hotfix `CASMTRIAGE-6715-noos-repo`
```suggestion
* Fixed an issue where an old version of the spire-agent was loaded into Nexus resulting in failed ARM compute builds

```

**Review by ndavidson-hpe — CHANGES_REQUESTED** [review]

File: upgrade/1.5.1/index.md (hunk 2 above)
Comment (ndavidson-hpe) [reply upgrade/1.5.1/index.md]: Made a comment with a suggested change to add the line for the spire-agent fix.

**Comment by rustydb** [comment]
We need to backport release/1.4:upgrade/scripts/upgrade/upload-ncn-images.sh (CASMTRIAGE-6851).
The script will need a small amendment to replace `1.4` with `1.5` https://github.com/Cray-HPE/docs-csm/blob/release/1.4/upgrade/scripts/upgrade/upload-ncn-images.sh#L38-L42

**Review by rustydb — CHANGES_REQUESTED** [review]
https://github.com/Cray-HPE/docs-csm/pull/4888#issuecomment-2051961020

**Review by mharding-hpe — APPROVED** [review]

File: upgrade/1.5.1/README.md [hunk 16]
```diff
@@ -0,0 +1,313 @@
...
+## Introduction
+
+This document guides an administrator through the patch update to Cray Systems Management `v1.5.1` from `1.5.0`.
+If upgrading from CSM `v1.4.x` directly to `v1.5.1`, follow the procedures described
+in [Upgrade CSM](../README.md) instead.
+
+## Bug fixes and improvements
+
+* Updated [IMS](../../glossary.md#image-management-service-ims) to allow for remote node builds
+* Support for Paradise hardware
```
Comment (rfrost-hpe) [inline upgrade/1.5.1/README.md]: Need to remove this from 1.5.1 docs.   Paradise support will be in 1.5.2

File: upgrade/1.5.1/README.md [hunk 17]
```diff
@@ -0,0 +1,312 @@
...
+* Fixed an issue where an old version of the spire-agent was loaded into Nexus resulting in failed ARM compute builds
+
+## Steps
+
+1. [Preparation](#preparation)
+1. [Setup Nexus](#setup-nexus)
+1. [Update Argo CRDs](#update-argo-crds)
+1. [Upgrade services](#upgrade-services)
+1. [Upload NCN images](#upload-ncn-images)
+1. [Upgrade Ceph and stop local Docker registries](#upgrade-ceph-and-stop-local-docker-registries)
```
Comment (leliasen-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
```

File: upgrade/1.5.1/README.md [hunk 18]
```diff
@@ -0,0 +1,312 @@
...
+
+This step will install the `smart-mon` rpm on storage nodes, and reconfigure the `node-exporter` to provide `smartmon`
+metrics.
+
+1. (`ncn-m001#`) Execute the following script.
+
+   ```bash
+   /usr/share/doc/csm/scripts/operations/ceph/enable-smart-mon-storage-nodes.sh
+   ```
+
```
Comment (leliasen-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
```

Remove the entire blocks 'upgrade ceph and stop local docker registries' and 'Enable smartmon on storage nodes'. Both of these are fixed in CSM 1.5 and don't need to be in the patch instructions

File: upgrade/1.5.1/README.md [hunk 19]
```diff
@@ -0,0 +1,312 @@
...
+## Steps
+
+1. [Preparation](#preparation)
+1. [Setup Nexus](#setup-nexus)
+1. [Update Argo CRDs](#update-argo-crds)
+1. [Upgrade services](#upgrade-services)
+1. [Upload NCN images](#upload-ncn-images)
+1. [Upgrade Ceph and stop local Docker registries](#upgrade-ceph-and-stop-local-docker-registries)
+1. [Enable `smartmon` metrics on storage NCNs](#enable-smartmon-metrics-on-storage-ncns)
```
Comment (leliasen-hpe) [inline upgrade/1.5.1/README.md]:
```suggestion
```

**Review by leliasen-hpe — CHANGES_REQUESTED** [review]

File: upgrade/1.5.1/README.md [hunk 20]
```diff
@@ -0,0 +1,312 @@
...
+* Fixed an issue where `backup_smd_postgres.sh` script is not executable
+* `CAST-34869`: Fixed an issue where `cray-hms-rts-snmp` would enter encounter a segmentation fault
+* `CAST-34141`: Fixed an issue where the `QLogic` driver would crash the system
+* Fixed an issue where an old version of the spire-agent was loaded into Nexus resulting in failed ARM compute builds
+
+## Steps
+
+1. [Preparation](#preparation)
+1. [Setup Nexus](#setup-nexus)
+1. [Update Argo CRDs](#update-argo-crds)
```
Comment (bo-quan) [inline upgrade/1.5.1/README.md]:
```suggestion
```

File: upgrade/1.5.1/README.md [hunk 21]
```diff
@@ -0,0 +1,312 @@
...
+
+```bash
+for c in $(kubectl get crd |grep argo | cut -d' ' -f1)
+do
+   kubectl label --overwrite crd $c app.kubernetes.io/managed-by="Helm"
+   kubectl annotate --overwrite crd $c meta.helm.sh/release-name="cray-nls"
+   kubectl annotate --overwrite crd $c meta.helm.sh/release-namespace="argo"
+done
+```
+
```
Comment (bo-quan) [inline upgrade/1.5.1/README.md]:
```suggestion
```
Remove the entire "Update Argo CRDs" block as it's fixed in CSM 1.5 and doesn't need to be in the patch instructions

**Review by bo-quan — CHANGES_REQUESTED** [review]

**Review by rustydb — APPROVED** [review]

**Review by rfrost-hpe — APPROVED** [review]

**Review by rustydb — APPROVED** [review]

**Pull request merged**